### PR TITLE
Master forum pimp and pump promenade loma

### PR DESCRIPTION
**Pull request by lm4649**
website_*

* = forum, profile, slides, slides_forum, slides_survey

Purpose
=======
Make it possible for users to have an overview and search in all forums. This includes searching for posts of other users, and their own upvoted posts, much like /slides/all.

Give users a better experience when browsing website profiles looking for courses, certifications, forum questions, ...

Specifications
=============
Add route for `/forum/all`

change templates in `website_forum` to show posts even if `forum` is not specified to shows results and search in all forums

Add search bars for Questions, Answers and Upvotes, small UI changes here and there.

Task-2676848
Co-authored by: Fabio Barbero <faba@odoo.com>

### CODE REVIEW

**Review by flch-odoo — COMMENTED** [review]
Just one comment left at this stage, in addition to the following: 
* Could you add a comment linking to the previous PR with the many comments so we can easily go back if necessary.
* A new rebase seems required.
Once it is done, please green the stage on my behalf :)

File: addons/website_forum/controllers/website_forum.py [hunk 1]
```diff
@@ -85,41 +85,55 @@ def _get_forum_port_search_options(self, forum=None, tag=None, filters=None, my=
...
+                 '/forum/<model("forum.forum"):forum>',
                  '/forum/<model("forum.forum"):forum>/page/<int:page>',
                  '''/forum/<model("forum.forum"):forum>/tag/<model("forum.tag"):tag>/questions''',
                  '''/forum/<model("forum.forum"):forum>/tag/<model("forum.tag"):tag>/questions/page/<int:page>''',
                  ], type='http', auth="public", website=True, sitemap=sitemap_forum)
-    def questions(self, forum, tag=None, page=1, filters='all', my=None, sorting=None, search='', **post):
+    def questions(self, forum=None, tag=None, page=1, filters='all', my=None, sorting=None, search='', author_id=False, include_answers=False, **post):
         Post = request.env['forum.post']
 
+        author = request.env['res.users'].browse(int(author_id)).exists()
```
Comment (tde-banana-odoo) [inline addons/website_forum/controllers/website_forum.py:102]: Is the exists necessary ? As it adds a query ... and if you give a wrong ID, well, is it that bad ?

File: addons/website_forum/views/website_profile_templates.xml [hunk 2]
```diff
@@ -57,10 +57,25 @@
                 <t t-set="is_profile" t-value="true"/>
                 <div role="tabpanel" class="o_wforum_profile_tab tab-pane" id="questions">
                     <div class="mb-4">
-                        <h5 class="border-bottom pb-1">
+                        <h5 class="border-bottom pb-1 d-flex align-items-center justify-content-between">
                             Questions
-                            <t t-call="website_forum.forum_filter_tag"/>
+                             <t t-if="count_questions &gt; 0">
+                                <t class="float-end" t-call="website.website_search_box_input">
```
Comment (tde-banana-odoo) [inline addons/website_forum/views/website_profile_templates.xml:63]: The two 't' tags can be nested normally

File: addons/website_forum/views/website_profile_templates.xml [hunk 3]
```diff
@@ -101,10 +116,26 @@
                 </div>
                 <div role="tabpanel" class="o_wforum_profile_tab tab-pane" id="answers">
                     <div class="mb-4">
-                        <h5 class="border-bottom pb-1">
+                        <h5 class="border-bottom pb-1 d-flex align-items-center justify-content-between">
                             Answers
-                            <t t-call="website_forum.forum_filter_tag"/>
+                             <t t-if="count_questions &gt; 0">
+                                <t class="float-end" t-call="website.website_search_box_input">
```
Comment (tde-banana-odoo) [inline addons/website_forum/views/website_profile_templates.xml:122]: Same here

File: addons/website_forum/views/website_profile_templates.xml [hunk 4]
```diff
@@ -127,16 +158,33 @@
                 </div>
                 <div role="tabpanel" class="o_wforum_profile_tab tab-pane" id="votes" t-if="uid == user.id">
                     <div class="mb-4">
-                        <h5 class="border-bottom pb-1">Votes</h5>
+                        <h5 class="border-bottom pb-1 flex align-items-center justify-content-between" style="height:43px;">
+                            Votes
+                            <t t-if="my_profile">
+                                <t class="float-end" t-call="website.website_search_box_input">
```
Comment (tde-banana-odoo) [inline addons/website_forum/views/website_profile_templates.xml:164]: And here ^^

File: addons/website_profile/views/website_profile.xml [hunk 5]
```diff
@@ -257,11 +257,12 @@
                                         <th><small class="fw-bold">Joined</small></th>
                                         <td><span t-field="user.create_date" t-options='{"format": "d MMM Y"}'/></td>
                                     </tr>
-                                    <tr>
-                                        <th><small class="fw-bold">Badges</small></th>
-                                        <td t-if="user.badge_ids" t-esc="len(user.badge_ids.filtered(lambda b: b.badge_id.website_published))"/>
-                                        <td t-else="">0</td>
-                                    </tr>
+                                     <t t-if="user.badge_ids">
+                                        <tr>
```
Comment (tde-banana-odoo) [inline addons/website_profile/views/website_profile.xml:261]: tr can be bested with t tag

**Review by tde-banana-odoo — COMMENTED** [review]
Hello, thanks for the PR ! Added some comments while passing by :)

Cheers !

File: addons/website_forum/views/forum_forum_templates_layout.xml [hunk 6]
```diff
@@ -18,15 +18,18 @@
 <!-- PAGE INDEX -->
 <!-- ============================================================ -->
 <template id="forum_model_nav">
+    <t t-set="_forum_slug" t-value="slug(forum) if forum else 'all'"/>
+    <t t-set="_all_forums"><t>All forums</t></t>
```
Comment (tde-banana-odoo) [inline addons/website_forum/views/forum_forum_templates_layout.xml:22]: Extra "t" is not necessary normally, is it ?

**Review by tde-banana-odoo — COMMENTED** [review]
Small comment :)

File: addons/website_forum/views/forum_forum_templates_layout.xml (hunk 6 above)
Comment (lm4649) [reply addons/website_forum/views/forum_forum_templates_layout.xml]: You are right. Extra "t" is useless. Removed.

**Comment by tde-banana-odoo** [comment]
Forgot to check PR name ^^

**Pull request closed without merging**